crates/rpc/rpc/src/eth/api/block.rs

```diff
@@ -15,6 +15,44 @@ use reth_rpc_types::{Index, RichBlock, TransactionReceipt};
 
 use reth_rpc_types_compat::block::{from_block, uncle_block_from_header};
 use reth_transaction_pool::TransactionPool;
+
+#[cfg(feature = "optimism")]
+use reth_primitives::U256;
+#[cfg(feature = "optimism")]
+use revm::L1BlockInfo;
+#[cfg(feature = "optimism")]
+use std::rc::Rc;
+
+/// Optimism Transaction Metadata
+///
+/// Includes the L1 fee and data gas for the tx along with the L1
+/// block info. In order to pass the [OptimismTxMeta] into the
+/// async colored `build_transaction_receipt_with_block_receipts`
+/// function, a reference counter for the L1 block info is
+/// used so the L1 block info can be shared between receipts.
+#[cfg(feature = "optimism")]
+#[derive(Debug, Default, Clone)]
+pub(crate) struct OptimismTxMeta {
+    /// The L1 block info.
+    pub(crate) l1_block_info: Option<Rc<L1BlockInfo>>,
```

Comment:
this Rc is very redundant because eventually these (copy) fields will all be consumed when constructing the receipt.

Reply:
Removed the `Rc` in https://github.com/anton-rs/op-reth/pull/155